enterprise/web.core/src/org/netbeans/modules/web/jspcompiler/SmapResolver.java

```diff
@@ -333,7 +334,7 @@ public Map getFileNames() {
      * get primary jsp filename
      */
     public String getPrimaryJspFileName() {
-        TreeMap tm = new TreeMap(fsection);
+        TreeMap tm = new TreeMap<>(fsection);
```

Comment:
`fsection` is a `HashTable<String,String>`, why not make the result a `TreeMap<String,String>`, type `o` correctly to `String` and remove the cast in line 339

Reply:
Good spot. Done!